compass/landice/tests/mismipplus/setup_mesh.py

```diff
@@ -0,0 +1,301 @@
+            (ds_mesh.yEdge == unique_ys_edge[-1]))
+    # WHL: zero out the edges on the boundary
+    #      (not necessary because velocity will also be zero)
+    ds_mesh['dvEdge'] = xr.where(mask, 0.0, ds_mesh.dvEdge)
+
+    # Boolean mask for the indexed of edges between N/S boundary cells
+    mask = ((ds_mesh.yEdge == unique_ys_edge[1]) |
+            (ds_mesh.yEdge == unique_ys_edge[-2]))
+    # cut length in half for edges between boundary cells
+    ds_mesh['dvEdge'] = xr.where(mask, ds_mesh.dvEdge * 0.5, ds_mesh.dvEdge)
+
+    return ds_mesh
+
+
+def _mismipplus_bed(x, y):
+    """
+    Equations 1-4 from Asay-Davis et al. (2016).
+
+    Parameters
+    ----------
+    x : xarray.DataArray
+        x cell coordinates
+
+    y : xarray.DataArray
+        y cell coordinates
+    """
+    x = x / 1.e3      # m to km
+    y = y / 1.e3      # m to km
+    B0 = -150.        # m
+    B2 = -728.8       # m
+    B4 = 343.91       # m
+    B6 = -50.57       # m
+    x_bar = 300.      # km
+    x_tilde = x / x_bar
+    dc = 500.         # m
+    fc = 4.           # km
+    wc = 24.          # km
+    Ly = 80.          # km
+    Bmax = -720.      # m
+
+    B_x = B0 + B2 * x_tilde**2 + B4 * x_tilde**4 + B6 * x_tilde**6
+    B_y = dc / (1 + np.exp(-2 * (y - Ly / 2 - wc) / fc)) +\
+        dc / (1 + np.exp(2 * (y - Ly / 2 + wc) / fc))
+
+    Bsum = B_x + B_y
+
+    z_b = np.maximum(Bsum, Bmax)  # Eqn 1. (Asay-Davis et al. 2016)
+
+    return z_b
+
+
+def _setup_MISMPPlus_IC(config, logger, filename):
+    """
+    Add the inital condition for the MISMIP+ spinup to the given MPAS mesh file
+
+    Parameters
+    ----------
+    config : comass.config.CompassConfigParser
+        Configuration options for this test case
+
+    logger : logging.Logger
+        A logger for output from the step
+
+    filename : str
+        NetCDF file to place the MISMIP+ ICs in
+
+    """
+
+    # Hard code some parameters from Table 1. of Asay-Davis et al. 2016
+    accum = 0.3       # m^{-1}
+    C = 3.160e6       # Pa m^{-1/3} s^{1/3}
+    rhoi = 918        # kg m^{-3}
+    spy = 31556926    # s a^{-1}
+    xcalve = 640.e3   # m
+
+    # Read parameters from the .cfg file
+    section = config['mesh']
+    nVertLevels = section.getint('levels')
+    init_thickness = section.getfloat('init_thickness')
+
+    # open the file
+    src = xr.open_dataset(filename)
+
+    # Use `.loc[:]` for setting the initial conditions  since we are setting
+    # the fields with scalar values. This ensures values are properly broadcast
+    # against the fields coordinates
+
+    # Set the bedTopography
+    src['bedTopography'].loc[:] = _mismipplus_bed(src.xCell, src.yCell)
+
+    # Set the ice thickness
+    src['thickness'].loc[:] = xr.where(src.xCell < xcalve, init_thickness, 0.)
+
+    # Convert SMB from m/yr to kg/m2/s
+    accum *= rhoi / spy
+    # Set the surface mass balance
+    src['sfcMassBal'].loc[:] = xr.where(src.xCell > xcalve, accum, -100.)
```

Comment:
The original implementation here of using strong negative SMB to maintain calving position is fine for now.  @trhille , if you use these meshes for evolving calving fronts, you'll want to remove this.  We could replace this method with `calvingMask` instead (which didn't exist when the original MISMIP+ test case was written).  I don't think that needs to happen in this PR, unless you think it would be timesaving to have that implementation switched now.

Reply:
Good catch. I think eventually we do want to replace this with `calvingMask`, but I don't care much whether that is done in this PR or in a short follow-up. I guess we might as well do it now, since I think it is an easy change.